src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,80 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user;
+
+    public RegistrationServiceImplTest() {
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("login");
+        user.setPassword("password");
+        user.setAge(25);
+    }
+
+    @Test
+    void register_NullUser_NotOk() {
+        assertThrows(RuntimeException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_NullLogin_NotOk() {
+        user.setLogin(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_LoginAlreadyExist_NotOk() {
+        User userToCheckLogin = new User();
+        userToCheckLogin.setLogin("loginForCheck");
+        userToCheckLogin.setPassword("password");
+        userToCheckLogin.setAge(18);
+        registrationService.register(userToCheckLogin);
+        assertThrows(RuntimeException.class, () -> registrationService.register(userToCheckLogin));
+    }
+
+    @Test
+    void register_NullPassword_NotOk() {
+        user.setPassword(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_PasswordLessThanSixCharacters_NotOk() {
+        user.setPassword("pass1");
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_NullAge_NotOk() {
+        user.setAge(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_AgeLessThan18_NotOk() {
+        user.setAge(17);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_userRegistered_Ok() {
+        StorageDao storageDao = new StorageDaoImpl();
```

Comment:
don't use StorageDaoImpl, register returns User, it's enough to check a user that you passed to register() method with the one that will be returned by it